Support symbolic for conv_tbc (#58359)

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #58697 [ONNX] Enable support for roll() op. (#58389)
* #58696 [ONNX] handle aten::_set_item on Dict in convertInplaceOpsAndTrackAlias (#58317)
* #58695 [ONNX] use consistent quoting for string literals (#57757)
* #58694 [ONNX] Improve lower tuples and handle control flow  (#57650)
* #58693 [ONNX] Update special post process for SequenceInsert after SequenceEmpty (#56965)
* **#58692 Support symbolic for conv_tbc (#58359)**
* #58691 [ONNX] RNN scripting (#57564)
* #58690 [ONNX] Update instance_norm2 symbolic to handle track_running_stats=True (#55051)

This is a fix for exporting fairseq models, see:
```python
model = torch.hub.load(github, 'conv.wmt14.en-fr', tokenizer='moses', bpe='subword_nmt')
model = torch.hub.load(github, 'conv.wmt17.en-de', tokenizer='moses', bpe='subword_nmt')
```
With this fix, and comment out model script one line `GradMultiply`, these two models can be exported successfully with perf met.

The original PR https://github.com/pytorch/pytorch/pull/57708 has merging issue, use this one instead.

Co-authored-by: David <jiafa@microsoft.com>

Differential Revision: [D28714809](https://our.internmc.facebook.com/intern/diff/D28714809)

jiafatom committed:
Support symbolic for conv_tbc (#58359)

This is a fix for exporting fairseq models, see:
```python
model = torch.hub.load(github, 'conv.wmt14.en-fr', tokenizer='moses', bpe='subword_nmt')
model = torch.hub.load(github, 'conv.wmt17.en-de', tokenizer='moses', bpe='subword_nmt')
```
With this fix, and comment out model script one line `GradMultiply`, these two models can be exported successfully with perf met.

The original PR https://github.com/pytorch/pytorch/pull/57708 has merging issue, use this one instead.

Co-authored-by: David <jiafa@microsoft.com>

[ghstack-poisoned]

BowenBao committed:
Update on "Support symbolic for conv_tbc (#58359)"


This is a fix for exporting fairseq models, see:
```python
model = torch.hub.load(github, 'conv.wmt14.en-fr', tokenizer='moses', bpe='subword_nmt')
model = torch.hub.load(github, 'conv.wmt17.en-de', tokenizer='moses', bpe='subword_nmt')
```
With this fix, and comment out model script one line `GradMultiply`, these two models can be exported successfully with perf met.

The original PR https://github.com/pytorch/pytorch/pull/57708 has merging issue, use this one instead.

Co-authored-by: David <jiafa@microsoft.com>

[ghstack-poisoned]

SplitInfinity commented:
@splitinfinity has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28622446).

SplitInfinity commented:
@SplitInfinity has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28714809).

BowenBao committed:
Update on "Support symbolic for conv_tbc (#58359)"


This is a fix for exporting fairseq models, see:
```python
model = torch.hub.load(github, 'conv.wmt14.en-fr', tokenizer='moses', bpe='subword_nmt')
model = torch.hub.load(github, 'conv.wmt17.en-de', tokenizer='moses', bpe='subword_nmt')
```
With this fix, and comment out model script one line `GradMultiply`, these two models can be exported successfully with perf met.

The original PR https://github.com/pytorch/pytorch/pull/57708 has merging issue, use this one instead.

Co-authored-by: David <jiafa@microsoft.com>

Differential Revision: [D28714809](https://our.internmc.facebook.com/intern/diff/D28714809)

[ghstack-poisoned]

SplitInfinity commented:
@SplitInfinity has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28714809).
